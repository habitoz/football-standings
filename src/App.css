@import url('https://fonts.googleapis.com/css2?family=Cabin&family=Lato:wght@700&display=swap');

:root {
  --borderColor: #e8e8e8;
  --darkColor: #1D1D1D;
  --primaryColor: #EB4B89;
  --blueText: #4386bf;
  --grayColor: #e8e8e8;
}
* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: 'Cabin', sans-serif;
}
.App {
  width: 100%;
  height: 100%;
  min-height: 100vh;
  display: flex;
  background-color: var(--darkColor);
  flex-direction: column;
  color: #fff;
}

.Main {
  padding: 0 0 0 0;
  flex: 1;
}

.span-2-col {
  grid-column: 1 / 3;
}

.ErrorMessage {
  color: red;
  font-weight: bold;
}

.text-center {
  text-align: center;
}
