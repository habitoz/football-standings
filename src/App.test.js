import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import App from './App';

const mockStore = configureMockStore([thunk]);

test('App component renders as expected', () => {
  const initialState = {
    league: {
      leagues: [
        {
          id: 1,
          name: 'English PL',
          images: { dark: '' },
        },
      ],
      searchString: '',
      error: null,
    },
    appInfo: {
      activePage: 'League',
      showSearchForm: true,
    },
  };

  const store = mockStore(initialState);

  const app = render(
    <MemoryRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </MemoryRouter>,
  );

  expect(app).toMatchSnapshot();
});
