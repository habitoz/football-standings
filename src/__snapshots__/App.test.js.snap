// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav
          class="Nav"
        >
          <a
            aria-current="page"
            class="Link AppLogo active"
            href="/"
          >
            <img
              alt=""
              class="MenuIcon"
              src="menu.svg"
            />
          </a>
          <p
            class="AppName"
          >
            League
          </p>
          <button
            class="searchBtn"
            type="button"
          >
            <i
              class="fa fa-search"
            />
          </button>
          <ul
            class="NavItems"
          />
        </nav>
        <main
          class="Main"
        >
          <div
            class="PageContainer"
          >
            <div>
              <input
                class="filterInput"
                type="text"
              />
            </div>
            <ul
              class="ListContainer"
            >
              <li>
                 
                <div
                  class="CardContainer"
                  role="navigation"
                >
                  <div
                    class="LeagueImage"
                    style="background-image: url(undefined);"
                  />
                  <p
                    class="LeagueName"
                  >
                    English PL
                  </p>
                </div>
                 
              </li>
            </ul>
          </div>
        </main>
        <footer
          class="FooterContainer"
        >
          <div>
            <p
              class="CopyRightText"
            >
              © 2023
            </p>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav
        class="Nav"
      >
        <a
          aria-current="page"
          class="Link AppLogo active"
          href="/"
        >
          <img
            alt=""
            class="MenuIcon"
            src="menu.svg"
          />
        </a>
        <p
          class="AppName"
        >
          League
        </p>
        <button
          class="searchBtn"
          type="button"
        >
          <i
            class="fa fa-search"
          />
        </button>
        <ul
          class="NavItems"
        />
      </nav>
      <main
        class="Main"
      >
        <div
          class="PageContainer"
        >
          <div>
            <input
              class="filterInput"
              type="text"
            />
          </div>
          <ul
            class="ListContainer"
          >
            <li>
               
              <div
                class="CardContainer"
                role="navigation"
              >
                <div
                  class="LeagueImage"
                  style="background-image: url(undefined);"
                />
                <p
                  class="LeagueName"
                >
                  English PL
                </p>
              </div>
               
            </li>
          </ul>
        </div>
      </main>
      <footer
        class="FooterContainer"
      >
        <div>
          <p
            class="CopyRightText"
          >
            © 2023
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
